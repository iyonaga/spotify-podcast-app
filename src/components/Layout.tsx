import SearchBar from './SearchBar';

interface Props {
  children: React.ReactNode;
}

const Layout: React.VFC<Props> = ({ children }) => {
  return (
    <div>
      <SearchBar />
      <main>{children}</main>
    </div>
  );
};

export default Layout;
