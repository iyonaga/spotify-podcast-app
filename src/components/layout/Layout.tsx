import SearchBar from './SearchBar';
import Sidebar from './Sidebar';

interface Props {
  children: React.ReactNode;
}

const Layout: React.VFC<Props> = ({ children }) => {
  return (
    <div className="flex min-h-screen">
      <Sidebar />
      <div className="px-[45px] pb-[40px] w-[calc(100%-250px)]">
        <SearchBar />
        <main className="mt-[40px]">{children}</main>
      </div>
    </div>
  );
};

export default Layout;
