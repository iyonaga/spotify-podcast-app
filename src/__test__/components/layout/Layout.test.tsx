import { render, screen } from '@testing-library/react';
import Layout from '@/components/layout/Layout';

jest.mock(
  '@/components/layout/Sidebar',
  () =>
    function mockSidebar() {
      return <div />;
    }
);
jest.mock(
  '@/components/layout/SearchBar',
  () =>
    function mockSearchBar() {
      return <div />;
    }
);

describe('components/layout/Layout.tsx', () => {
  test('スナップショットが一致する', () => {
    const { asFragment } = render(<Layout>test</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('子要素が描画される', () => {
    const children = 'test';
    render(<Layout>{children}</Layout>);
    expect(screen.getByText(children)).toBeInTheDocument();
  });
});
