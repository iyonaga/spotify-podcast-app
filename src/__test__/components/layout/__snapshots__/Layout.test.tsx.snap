// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/Layout.tsx スナップショットが一致する 1`] = `
<DocumentFragment>
  <div
    class="flex min-h-screen"
  >
    <div />
    <div
      class="px-[45px] pb-[40px] w-[calc(100%-250px)]"
    >
      <div />
      <main
        class="mt-[40px]"
      >
        test
      </main>
    </div>
  </div>
</DocumentFragment>
`;
